metrics-operator/controllers/analysis/controller.go

```diff
@@ -0,0 +1,124 @@
+	Log        logr.Logger
+	MaxWorkers int //maybe 2 or 4 as def
+	Namespace  string
+	NewWorkersPoolFactory
+	common.IAnalysisEvaluator
+}
+
+//+kubebuilder:rbac:groups=metrics.keptn.sh,resources=analyses,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=metrics.keptn.sh,resources=analyses/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=metrics.keptn.sh,resources=analyses/finalizers,verbs=update
+// +kubebuilder:rbac:groups=metrics.keptn.sh,resources=keptnmetricsproviders,verbs=get;list;watch;
+//+kubebuilder:rbac:groups=metrics.keptn.sh,resources=analysisdefinitions,verbs=get;list;watch;
+//+kubebuilder:rbac:groups=metrics.keptn.sh,resources=analysisvaluetemplates,verbs=get;list;watch;
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// For more details, check Reconcile and its AnalysisResult here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (a *AnalysisReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	a.Log.Info("Reconciling Analysis")
+	analysis := &metricsapi.Analysis{}
+
+	//retrieve analysis
+	if err := a.Client.Get(ctx, req.NamespacedName, analysis); err != nil {
+		if errors.IsNotFound(err) {
+			// taking down all associated K8s resources is handled by K8s
+			a.Log.Info("Analysis resource not found. Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
+		a.Log.Error(err, "Failed to get the Analysis")
+		return ctrl.Result{}, nil
+	}
+
+	//check status, is there any error?/is empty?
+
+	//find AnalysisDefinition to have the collection of Objectives
+	analysisDef := &metricsapi.AnalysisDefinition{}
+	if analysis.Spec.AnalysisDefinition.Namespace == "" {
+		analysis.Spec.AnalysisDefinition.Namespace = a.Namespace
+	}
+	err := a.Client.Get(ctx,
+		types.NamespacedName{
+			Name:      analysis.Spec.AnalysisDefinition.Name,
+			Namespace: analysis.Spec.AnalysisDefinition.Namespace},
+		analysisDef,
+	)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			a.Log.Info(err.Error() + ", ignoring error since object must be deleted")
+			return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+		a.Log.Error(err, "Failed to retrieve the AnalysisDefinition")
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	//create multiple workers handling the Objectives
+	ctx, wp := a.NewWorkersPoolFactory(ctx, analysis, analysisDef, a.MaxWorkers, a.Client, a.Log, a.Namespace)
+
+	res, err := wp.DispatchAndCollect(ctx)
+	if err != nil {
+		a.Log.Error(err, "Failed to Collect all SLOs")
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	eval := a.Evaluate(res, analysisDef)
+	analysisResultJSON, err := json.Marshal(eval)
+	if err != nil {
+		a.Log.Error(err, "Could not marshal status")
+	}
+	analysis.Status = string(analysisResultJSON) //TODO add struct for status
```

Comment:
Can we add a field for the analysisResult in the status, i.e. `status.analysisResult`?

Reply:
this should be the overall result